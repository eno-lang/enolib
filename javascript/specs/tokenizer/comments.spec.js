const { inspectTokenization } = require('./util.js');

const input = '> value\n' +
              '> more value  \n' +
              '    >    value\n' +
              '    >    ';

describe('Comment tokenization', () => {
  it('performs as specified', () => {
    expect(inspectTokenization(input)).toMatchSnapshot();
  });
});
