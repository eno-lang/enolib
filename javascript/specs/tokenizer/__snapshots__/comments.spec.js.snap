// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment tokenization performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [],
    "type": Symbol(Document),
  },
  "_lineCount": 4,
  "_meta": Array [
    Object {
      "comment": "value",
      "line": 0,
      "ranges": Object {
        "comment": Array [
          2,
          7,
        ],
        "commentOperator": Array [
          0,
          1,
        ],
        "line": Array [
          0,
          7,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "more value",
      "line": 1,
      "ranges": Object {
        "comment": Array [
          10,
          20,
        ],
        "commentOperator": Array [
          8,
          9,
        ],
        "line": Array [
          8,
          22,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "value",
      "line": 2,
      "ranges": Object {
        "comment": Array [
          32,
          37,
        ],
        "commentOperator": Array [
          27,
          28,
        ],
        "line": Array [
          23,
          37,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": null,
      "line": 3,
      "ranges": Object {
        "commentOperator": Array [
          42,
          43,
        ],
        "line": Array [
          38,
          47,
        ],
      },
      "type": Symbol(Comment),
    },
  ],
}
`;
